en/test_and_ci/integration_testing.md

```diff
@@ -32,113 +34,113 @@ make px4_sitl_default sitl_gazebo
-:::
-
-####1.) Create a new test script
-
-Test scripts are located in `integrationtests/python_src/px4_it/mavros/`.
-See other existing scripts for examples. Also please consult the official ROS documentation on how to use [unittest](http://wiki.ros.org/unittest).
-
-
-Empty test skeleton:
-
-```python
-#!/usr/bin/env python
-# [... LICENSE ...]
-
-#
-# @author Example Author <author@example.com>
-#
-PKG = 'px4'
-
-import unittest
-import rospy
-import rosbag
-
-from sensor_msgs.msg import NavSatFix
-
-class MavrosNewTest(unittest.TestCase):
-    """
-    Test description
-    """
-
-    def setUp(self):
-        rospy.init_node('test_node', anonymous=True)
-        rospy.wait_for_service('mavros/cmd/arming', 30)
-
-        rospy.Subscriber("mavros/global_position/global", NavSatFix, self.global_position_callback)
-        self.rate = rospy.Rate(10) # 10hz
-        self.has_global_pos = False
-
-    def tearDown(self):
-        pass
-
-    #
-    # General callback functions used in tests
-    #
-    def global_position_callback(self, data):
-        self.has_global_pos = True
-
-    def test_method(self):
-        """Test method description"""
-
-        # FIXME: hack to wait for simulation to be ready
-        while not self.has_global_pos:
-            self.rate.sleep()
-
-        # TODO: execute test
-
-if __name__ == '__main__':
-    import rostest
-    rostest.rosrun(PKG, 'mavros_new_test', MavrosNewTest)
-```
-
-#### 2.) Run the new test only
+The **.test** files launch the corresponding Python tests defined in `integrationtests/python_src/px4_it/mavros/`
 
-```sh
-# Start simulation
-cd <PX4-Autopilot_clone>
-source Tools/setup_gazebo.bash
-roslaunch launch/mavros_posix_sitl.launch
 
-# Run test (in a new shell):
-cd <PX4-Autopilot_clone>
-source Tools/setup_gazebo.bash
-rosrun px4 mavros_new_test.py
-```
+## Write a New MAVROS Test (Python)
 
-#### 3.) Add new test node to a launch file
+This section explains how to write a new python test using ROS(1)/MAVROS, test it, and add it to the PX4 test suite.
 
-In `test/` create a new `<test_name>.test` ROS launch file.
-Call the test file using one of the base scripts *rostest_px4_run.sh* or *rostest_avoidance_run.sh*
+We recommend you review the existing tests as examples/inspiration (**integrationtests/python_src/px4_it/mavros/**).
+The official ROS documentation also contains information on how to use [unittest](http://wiki.ros.org/unittest) (on which this test suite is based).
 
-#### 4.) (Optional) Create a new target in the Makefile
-1. Open the Makefile
-2. Search the *Testing* section
-3. Add a new target name and call the test
+:::note
+More streamlined support for testing is planned (helper classes/methods etc.).
+:::
 
-Example:
-```sh
-tests_<new_test_target_name>: rostest
-	@"$(SRC_DIR)"/test/rostest_px4_run.sh mavros_posix_tests_<new_test>.test
-```
+To write a new test:
+
+1. Create a new test script by copying the empty test skeleton below:
```

Comment:
Where should this file be copied?

Later on I am going to want it to be run with the tests - do I have to add it to a makefile? How do I do that?